modules/shared/src/main/scala/org/tessellation/currency/schema/currency.scala

```diff
@@ -80,7 +80,8 @@ object currency {
     lastSnapshotHash: Hash,
     blocks: SortedSet[BlockAsActiveTip[CurrencyBlock]],
     tips: SnapshotTips,
-    info: CurrencySnapshotInfo
+    info: CurrencySnapshotInfo,
+    version: SemVer = SemVer.Empty
```

Comment:
I wonder. Shouldn't it be hardcoded/baked into the class upon definition instead of parametrizable?

Reply:
I baked more or less. Take a look

Reply:
ofc assuming that we do not need history in form of ADT